pkg/nsm/apis/common/common.proto

```diff
@@ -41,9 +41,20 @@ enum LocalMechanismType {
     SRIOV_INTERFACE = 4;
     HW_INTERFACE = 5;
 }
-
+// RemoteMechanism - Mechanism for connecting to a remote NSE
+// A RemoteMechanism may be partially specified, to communicate preferences
+// to a peer when negotiating a RemoteConnection, or may be fully specified
+// to indicate the outcome of that negotiation.
 message RemoteMechanism {
+    // RemoteMechanismType - type of RemoteMechanism
     RemoteMechanismType type = 1;
+    // parameters - parameters for the RemoteMechanism
+    // Known Keys:
+    //     parameters[src_ip] - ip used on by source NSM for the mechanism, specified by source NSM
+    //     parameters[dst_ip] - ip used on by destination NSM for the mechanism, specified by destination NSM
+    //     parameters[vnis] - comma seperated list of acceptable vnis for VXLAN, typically specified by source NSM
+    //                        Example: vnis=10-20,50-100
+    //     parameters[vni]  - actual vni used in fully specified VXLAN RemoteMechanism, typically specified by destination NSM
```

Comment:
It seems like this is where we're putting protocol specific items, so do you envision this parameters map growing large? It seems it would contain keys for protocol specific information for each RemoteMechanismType, am I right?

Reply:
@mestery Yes... some things (src_ip, dst_ip) I expect to be used commonly across many (though not all) RemoteMechanisms.